pkg/costmodel/costmodel.go

```diff
@@ -1075,27 +1075,38 @@ func (cm *CostModel) GetNodeCost(cp costAnalyzerCloud.Provider) (map[string]*cos
 		// not all providers are guaranteed to use this, so don't overwrite a Provider assignment if we can't find something under that capacity exists
 		gpuc := 0.0
 		q, ok := n.Status.Capacity["nvidia.com/gpu"]
+		gpuCount := q.Value()
 		_, hasReplicas := n.Labels["nvidia.com/gpu.replicas"]
 
-		if ok && !hasReplicas {
-			gpuCount := q.Value()
-			if gpuCount != 0 {
-				newCnode.GPU = fmt.Sprintf("%d", gpuCount)
-				newCnode.VGPU = newCnode.GPU
-				gpuc = float64(gpuCount)
-			}
-		} else if hasReplicas { // See https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-sharing.html
-			if q.Value() == 0 {
-				q = n.Status.Capacity["nvidia.com/gpu.shared"]
-			}
+		if ok && gpuCount != 0 && !hasReplicas {
+			newCnode.GPU = fmt.Sprintf("%d", gpuCount)
+			newCnode.VGPU = newCnode.GPU
+			gpuc = float64(gpuCount)
+		} else if hasReplicas {
+			// Special case. The "nvidia.com/gpu.replicas" label is usually
+			// applied by NVIDIA's GPU Operator or GPU Feature Discovery pod.
```

Comment:
It's always applied by GFD. The operator may optionally deploy GFD but it is (now) otherwise deployed through the device plugin.

Reply:
Comment is updated! Marking as resolved.